PubSub/src/Subscription.php

```diff
@@ -399,11 +408,42 @@ public function create(array $options = [])
         }
 
         $options = $this->formatSubscriptionDurations($options);
+        $options = $this->formatDeadLetterPolicyForApi($options);
 
-        $this->info = $this->connection->createSubscription([
-            'name' => $this->name,
-            'topic' => $this->topicName
-        ] + $this->formatDeadLetterPolicyForApi($options));
+        // convert optional args to protos
+        if (isset($options['expirationPolicy'])) {
+            $options['expirationPolicy'] = $this->requestHandler->getSerializer()->decodeMessage(
+                new ExpirationPolicy(),
+                $options['expirationPolicy']
+            );
+        }
+
+        if (isset($options['messageRetentionDuration'])) {
+            $options['messageRetentionDuration'] = new ProtobufDuration(
+                $this->formatDurationForApi($options['messageRetentionDuration'])
+            );
+        }
+
+        if (isset($options['retryPolicy'])) {
+            $options['retryPolicy'] = $this->requestHandler->getSerializer()->decodeMessage(
+                new RetryPolicy,
+                $options['retryPolicy']
+            );
+        }
+
+        if (isset($options['deadLetterPolicy'])) {
+            $options['deadLetterPolicy'] = $this->requestHandler->getSerializer()->decodeMessage(
+                new DeadLetterPolicy,
+                $options['deadLetterPolicy']
+            );
+        }
+
+        $this->info = $this->requestHandler->sendRequest(
+            SubscriberClient::class,
+            'createSubscription',
+            [$this->name,$this->topicName],
+            $options
+        );
```

Comment:
We should move these to the V2 implementation, which would change this significantly because we eed to construct a `CreateSubscriptionRequest`:
```suggestion
        $this->info = $this->requestHandler->sendRequest(
            SubscriberClient::class,
            'createSubscription',
            [$this->name,$this->topicName],
            $request
        );
```

Reply:
Done. Moved all calls to V2 GAPIC surface.